File: src/main/java/com/werfad/ConfigUI.java
package com.werfad;

import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class ConfigUI {
    private JPanel rootPanel;
    private JTextField hit1ColorTF;
    private JTextField hit2ColorTF0;
    private JTextField hit2ColorTF1;
    private JTextField charactersTF;
    private JTextField bgTF;
    private JCheckBox smartcaseCheckBox;

    public JPanel getRootPanel() {
        return rootPanel;
    }

    public String getCharacters() {
        return charactersTF.getText();
    }

    public void setCharacters(String s) {
        charactersTF.setText(s);
    }

    @Nullable
    Integer getHit1FontColor() {
        try {
            return Integer.parseUnsignedInt(hit1ColorTF.getText(), 16);
        } catch (NumberFormatException e) {
            return null;
        }
    }

    public void setHit1FontColor(int c) {
        hit1ColorTF.setText(Integer.toHexString(c));
    }

    @Nullable
    Integer[] getHit2FontColors() {
        try {
            return new Integer[]{
                    Integer.parseUnsignedInt(hit2ColorTF0.getText(), 16),
                    Integer.parseUnsignedInt(hit2ColorTF1.getText(), 16)};
        } catch (NumberFormatException e) {
            return null;
        }
    }

    public void setHit2FontColors(int c0, int c1) {
        hit2ColorTF0.setText(Integer.toHexString(c0));
        hit2ColorTF1.setText(Integer.toHexString(c1));
    }

    @Nullable
    public Integer getBgColor() {
        try {
            return Integer.parseUnsignedInt(bgTF.getText(), 16);
        } catch (NumberFormatException e) {
            return null;
        }
    }

    public void setBgColor(int c) {
        bgTF.setText(Integer.toHexString(c));
    }

    public Boolean isSmartcase() {
        return smartcaseCheckBox.isSelected();
    }

    public void setSmartcase(boolean caseSensitive) {
        smartcaseCheckBox.setSelected(caseSensitive);
    }
}

File: src/main/kotlin/com/werfad/ConfigUI.kt
package com.werfad

import javax.swing.JPanel
import javax.swing.JTextField

class ConfigUI {
    lateinit var rootPanel: JPanel
    private lateinit var hit1ColorTF: JTextField
    private lateinit var hit2ColorTF0: JTextField
    private lateinit var hit2ColorTF1: JTextField
    private lateinit var charactersTF: JTextField
    private lateinit var bgTF: JTextField

    var characters: String?
        get() = charactersTF.text
        set(s) {
            charactersTF.text = s
        }

    var hit1FontColor: Int
        get() = try {
            Integer.parseUnsignedInt(hit1ColorTF.text, 16)
        } catch (e: NumberFormatException) {
            0
        }
        set(c) {
            hit1ColorTF.text = Integer.toHexString(c)
        }

    var hit2FontColors: IntArray?
        get() = try {
            intArrayOf(
                Integer.parseUnsignedInt(hit2ColorTF0.text, 16),
                Integer.parseUnsignedInt(hit2ColorTF1.text, 16)
            )
        } catch (e: NumberFormatException) {
            null
        }
        set(arr) {
            arr?.let {
                hit2ColorTF0.text = Integer.toHexString(it[0])
                hit2ColorTF1.text = Integer.toHexString(it[1])
            }
        }

    var bgColor: Int
        get() = try {
            Integer.parseUnsignedInt(bgTF.text, 16)
        } catch (e: NumberFormatException) {
            0
        }
        set(c) {
            bgTF.text = Integer.toHexString(c)
        }
}